Create Longform theme article page and style hero section

gabrieltanchen opened the pull request:
#### Description of changes
resolves CIGIHub/cigi-tickets#267

![Screen Shot 2020-12-15 at 2 11 25 PM](https://user-images.githubusercontent.com/10100465/102270598-2e9a4c00-3eec-11eb-9d9e-0822d8b3de30.png)

johnbxu commented:
![image](https://user-images.githubusercontent.com/40705848/102434218-5370eb00-3fe2-11eb-963a-9c7a65e7e201.png)

- Hero section has different theming on smaller screens - font size, image overlay, social icons less spaced out.

johnbxu commented:
![image](https://user-images.githubusercontent.com/40705848/102434326-8c10c480-3fe2-11eb-8497-65e553aae3fb.png)

Block quote margins should be decreased

johnbxu commented on templates/themes/longform/article_page.html:
```diff
@@ -0,0 +1,65 @@
+      {% include_block block %}
+    {% endfor %}
+  </section>
+{% endblock %}
```
Template should include opinion disclaimer and about the author sections.

johnbxu commented on articles/models.py:
```diff
@@ -227,6 +227,12 @@ class HeroTitlePlacements(models.TextChoices):
 
+    def get_template(self, request, *args, **kwargs):
+        standard_template = super(ArticlePage, self).get_template(request, *args, **kwargs)
+        if self.theme and self.theme.name == 'Longform':
```
Not urgent, but this should probably automatically grab the corresponding template path based on theme name so it doesn't have a whole bunch of `if...else` s

gabrieltanchen commented:
@johnbxu Updated - good catch on the mobile hero.

Updated the `ThemeablePageAbstract` model to dynamically get a template theme directory.

Updated the block quote margins in the general block quote template - the margins were a bit too large across the board. The Longform theme margin is still a bit bigger than what's on production, but I think it's better spaced out.

Also added the disclaimer and author sections. There's another issue to style the author section - this issue was just focused on the hero.

johnbxu commented:
Somehow the subtitle is now displaying a `p` tag: 
![image](https://user-images.githubusercontent.com/40705848/102441734-d6993d80-3ff0-11eb-9a74-a2a988db2ac8.png)

johnbxu commented on templates/streams/block_quote_block.html:
```diff
@@ -1,34 +1,32 @@
 {% load wagtailcore_tags wagtailimages_tags %}
 
-<div class="stream-block stream-block-blockquote">
-  <div class="container-sm flex-row">
```
Removing the `flex-row` makes the image stack above the rest: 
![image](https://user-images.githubusercontent.com/40705848/102441795-fe88a100-3ff0-11eb-9505-6077e0de6597.png)

Also, should I change the other stream-blocks so the container class is on the same element as the `stream-block-x`?

gabrieltanchen commented:
@johnbxu This is updated and ready for another review.

I updated the block quote to use flex and use the medium container if there's an image. There's only one node in Drupal that uses the block quote images and it uses the medium container: https://www.cigionline.org/articles/can-g20-save-globalizations-waning-reputation
![Screen Shot 2020-12-17 at 11 08 27 AM](https://user-images.githubusercontent.com/10100465/102512751-60c8bc80-4058-11eb-9289-89194b6975f4.png)

The `<p>` elements in the subtitle is a trickier one. That must have been because you saved the page. In Drupal this data is entered just as a character input, but in Wagtail I implemented it as a `RichTextField`, so that editors can add things like italics, etc. So the data in Wagtail _should_ have the `<p>` tag surrounding the text, which is what got updated when the page is saved. I updated the CSS to assume that the `<p>` tag is there. I'll have to update the migrator to insert the surrounding `<p>` tags when migrating from a plain text field.
![Screen Shot 2020-12-17 at 11 08 03 AM](https://user-images.githubusercontent.com/10100465/102512969-a1c0d100-4058-11eb-86b3-824cadcdb2a6.png)
